spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/Merge.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.nn.{CAddTable, CAveTable, CMaxTable, CMulTable, CosineDistance, DotProduct, JoinTable, ParallelTable, Sequential => TSequential}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{MultiShape, Shape}
+
+import scala.reflect.ClassTag
+
+class Merge[T: ClassTag](
+   val layers: Array[AbstractModule[Activity, Activity, T]] = null,
+   val mode: String = "sum",
+   val concatAxis: Int = -1,
+   // MultiShape isn't directly supported for serialization. Use Shape instead.
+   val inputShape: Shape = null)(implicit ev: TensorNumeric[T])
+  extends KerasLayer[Tensor[T], Tensor[T], T](Merge.calcBatchInputShape(inputShape, layers)) {
+
+  private val mergeMode = mode.toLowerCase()
+  private var axis = concatAxis
+
+  require(mergeMode == "sum" || mergeMode == "mul" || mergeMode == "concat" || mergeMode == "ave"
+  || mergeMode == "cos" || mergeMode == "dot" || mergeMode == "max",
+  s"Invalid merge mode: $mergeMode")
+  require(layers.length >= 2, s"Merge must have at least two input layers " +
+    s"but found ${layers.length}")
+
+  private def computeOutputShapeForConcat(input: List[Shape], input1: Array[Int]): Shape = {
+    import scala.util.control.Breaks._
+    val output = input1.clone()
+    require(Math.abs(concatAxis) < output.length, s"Invalid concat axis $concatAxis")
+    axis = if (concatAxis < 0) concatAxis + output.length else concatAxis
+    var i = 1
+    while (i < input.length) {
+      val input_i = input(i).toSingle().toArray
+      var j = 0
+      while (j < input_i.length) {
+        if (j != axis) require(input_i(j)==output(j), s"Incompatible input dimension for merge " +
+          s"mode concat: (${output.deep.mkString(", ")}), " +
+          s"(${input_i.deep.mkString(", ")})")
+        j += 1
+      }
+      if (output(axis) == -1 || input_i(axis) == -1) {
+        output(i) = -1
+        break
+      }
+      output(axis) = output(axis) + input_i(axis)
+      i += 1
+    }
+    Shape(output)
+  }
+
+  private def validateInputDim(input: List[Shape], input1: Array[Int]): Unit = {
```

Comment:
give it a meaningful name ? i.e `sameInputShape` and also remove `input1` from  the parameter list

Reply:
OKOK. I forgot modifying this....